### PR TITLE
Check the number of type parameters before handling the Range type

### PR DESCRIPTION
**Pull request by todays-mitsui**
There is an issue where a user-defined `Range` type is incorrectly treated as `std::ops::Range`.  
The behavior differs depending on whether the type has exactly one type parameter or not, but in either case, the user-defined `Range` is ignored.  

**Example: A panic occurs when there are no type parameters.**  

```rust
use tsify_next::Tsify;

#[derive(Tsify)]
pub struct Range {}

#[derive(Tsify)]
//       ^^^^^ error: proc-macro derive panicked
//             help: message: index out of bounds: the len is 0 but the index is 0
pub struct A {
    range: Range,
}
```

When the `Range` type is specified for a field in a struct, Tsify expects it to have one type parameter.  
However, a user-defined Range type without type parameters causes a panic.

**Example: When there is one type parameter, it is treated as `std::ops::Range`.**  

```rust
use tsify_next::Tsify;

#[derive(Tsify)]
pub struct Range<T> {
    _phantom: std::marker::PhantomData<T>,
}

#[derive(Tsify)]
pub struct A {
    range: Range<()>,
}
```

Generated `.d.ts` file:  

```typescript
export interface Range {}

export interface A {
    range: { start: null; end: null };
    // Expected: `range: Range;`
}
```

This issue arises because the system cannot correctly determine whether `Range` refers to a user-defined type or `std::ops::Range`.  
Unfortunately, avoiding this issue when there is exactly one type parameter seems difficult.  
On the other hand, for cases with a different number of type parameters, the distinction is straightforward. Clearly, `Range` or `Range<T, U>` should not be interpreted as `std::ops::Range`.  

Adding a guard to `TsType::from_name()` to handle cases with exactly one type parameter may provide a solution to this problem.

### CODE REVIEW

File: tsify-next-macros/src/typescript/ts_type_from_name.rs [hunk 1]
```diff
@@ -117,7 +117,7 @@ impl TsType {
                 nanos_since_epoch: Self::NUMBER;
             },
 
-            "Range" | "RangeInclusive" => {
+            "Range" | "RangeInclusive" if args.len() == 1 => {
```
Comment (siefkenj) [inline tsify-next-macros/src/typescript/ts_type_from_name.rs:122]: Can you add a comment here explaining why we do the args check?

Comment (todays-mitsui) [reply tsify-next-macros/src/typescript/ts_type_from_name.rs:122]: https://github.com/siefkenj/tsify/pull/45/commits/d9c0a866d8ec4831dabe9a5ab326738437b675ff
Comment added.

**Comment by siefkenj** [comment]
Hmm. Looks like the `expandtest` tests are failing. Possibly because they are using a newer rust version than when they were last run...would you mind rebuilding those with `MACROTEST=overwrite cargo test`?

**Comment by todays-mitsui** [comment]
I ran `MACROTEST=overwrite cargo test` and committed the changes in the `tests/expand/` directory.
https://github.com/siefkenj/tsify/pull/45/commits/ee02defabfa71dc4958d2f3c74d74d854646064b

I also ran `./test.sh` locally, and it seems that all the tests passed.

File: tests/expand/borrow.expanded.rs
Comment (todays-mitsui) [inline tests/expand/borrow.expanded.rs:1]: This file was automatically updated by running `MACROTEST=overwrite cargo test`.

File: tests/expand/generic_enum.expanded.rs
Comment (todays-mitsui) [inline tests/expand/generic_enum.expanded.rs:1]: This file was automatically updated by running `MACROTEST=overwrite cargo test`.

File: tests/expand/generic_struct.expanded.rs
Comment (todays-mitsui) [inline tests/expand/generic_struct.expanded.rs:1]: This file was automatically updated by running `MACROTEST=overwrite cargo test`.

**Comment by siefkenj** [comment]
Very weird. There are new tests failing. Do they fail on your system?

**Comment by todays-mitsui** [comment]
Looking at the workflow logs, it seems the test is failing during the execution of `wasm-pack test --node`.
On my local environment, this test passed without issues.

I tried a solution in the forked repository.
It appears that the default version of Node on ubuntu-latest is v18.20.5, which seems outdated.
By upgrading the Node version to 20, `wasm-pack test --node` passes successfully.

**Comment by todays-mitsui** [comment]
Oops, it seems the e2e test is still failing even after upgrading the Node version.
This fix looks straightforward—perhaps a mistake caused some spaces in the snapshot to be removed?

I'll create a Pull Request to address this workflow issue.

**Comment by todays-mitsui** [comment]
I have created a Pull Request to fix the workflow. Please review it.

- https://github.com/siefkenj/tsify/pull/47

Once the workflow completes without errors, it should be easier to review and accept this Pull Request.

**Comment by siefkenj** [comment]
Thanks for this! And thank you for adding tests right from the start :-D

**Pull request merged**